### PR TITLE
Track typename fix

### LINKED ISSUE #203: Update Event Selection To scroll to element
On large datasets the selected item might not be in the current scrolling location of the event chart.  We should probably scroll to center or attempt to center the item when it is selected.

### LINKED ISSUE #229: Editing the name of a category in the types dropdown doesn't edit it in all the tracks dropdown
I think on the backend it makes the change, though all of the types listed in the tracks coumn still have the old names

### LINKED ISSUE #235: Add ability to turn timeline view on and off

### PR DESCRIPTION
**Pull request by BryonLewis**
Track Typename fix and some other minor issues I'm trying to tie together into this PR.

Fixes #229 - Added in a watcher to the Type Name for the TrackItem so it can update properly when changed externally
Fixes #203 - Fixed issue where it now scrolls to the proper item.  During the process I noticed that there were issues with the current frame indicator going over the top of the scrollbar and the scrollbar triggering frame changes as well.  I've updated to add a margin around the Detection/Event viewers so that way we can have scroll bars.  This also has the benefit of cleaning up the Numbers for the Detection view.
image
image

Want to add to this:
Fixes #235 - Changed the button into two independent buttons with a toggle.  You can toggle between the two and then if both are off it will hide the timeline view.  Also added to the `Image-Annotator` and `Video-Annotator` that it will resize itself anytime the size of the panel changes.  This makes sure that the image displayed will take up the full area.
image
image

### CODE REVIEW

File: client/src/components/annotators/ImageAnnotator.vue [hunk 1]
```diff
@@ -262,7 +276,11 @@ export default {
         </v-progress-circular>
       </div>
     </div>
-    <slot name="control" />
+    <slot
+      ref="control"
+      name="control"
+      @resize="onResize"
```
Comment (subdavis) [inline client/src/components/annotators/ImageAnnotator.vue:282]: I'm pretty sure you [can't do that](https://github.com/vuejs/vue/issues/4781#issuecomment-274820814)

I actually don't see where any component in our code emits an event named 'resize'

File: client/src/components/annotators/ImageAnnotator.vue [hunk 2]
```diff
@@ -38,7 +39,20 @@ export default {
...
+        this.onResize();
+      });
+      this.observer.observe(this.$refs.container);
+    }
+  },
+  beforeDestroy() {
+    if (this.observer) {
+      this.observer.unobserve(this.$refs.container);
+    }
+  },
```
Comment (subdavis) [inline client/src/components/annotators/ImageAnnotator.vue:55]: Now that you're using the more robust `ResizeObserver` should the `v-resize` event from Vuetify be dropped from the template?

File: client/src/components/annotators/VideoAnnotator.vue [hunk 3]
```diff
@@ -33,7 +33,20 @@ export default {
...
+        this.onResize();
+      });
+      this.observer.observe(this.$refs.container);
+    }
+  },
+  beforeDestroy() {
+    if (this.observer) {
+      this.observer.unobserve(this.$refs.container);
+    }
+  },
```
Comment (subdavis) [inline client/src/components/annotators/VideoAnnotator.vue:49]: These are identical, so you might consider moving them to the `annotator` mixin?

File: client/src/components/controls/EventChart.vue [hunk 4]
```diff
@@ -209,11 +226,12 @@ export default {
       if (remainder > 10) {
         return;
       }
-      const top = offsetY - (offsetY % 15);
+      const top = offsetY - (offsetY % 15) + 3;
```
Comment (subdavis) [inline client/src/components/controls/EventChart.vue:229]: I'd complain about magic numbers, but this file is pretty much only magic numbers.  We should take care of that at some point, but now is not that time.

File: client/src/components/annotators/ImageAnnotator.vue (hunk 1 above)
Comment (BryonLewis) [reply client/src/components/annotators/ImageAnnotator.vue:282]: yup that looks like a mistake that I left in there when I was trying something else before utilizing the `ResizeObserver`

File: client/src/components/annotators/ImageAnnotator.vue (hunk 2 above)
Comment (BryonLewis) [reply client/src/components/annotators/ImageAnnotator.vue:55]: thank you, don't know why I glossed over the fact that `annotator` was a mixin for each of these components.

File: client/src/components/annotators/VideoAnnotator.vue (hunk 3 above)
Comment (BryonLewis) [reply client/src/components/annotators/VideoAnnotator.vue:49]: thank you, don't know why I glossed over the fact that `annotator` was a mixin for each of these components.

**Comment by BryonLewis** [comment]
Going to add in the dedicated button and transitions next.  Was just addressing some of the issues.

**Comment by subdavis** [comment]
On second thought, Leaving it with no transition is totally fine.  I don't know what came over me

**Comment by BryonLewis** [comment]
Added in the Collapse button and swapped the Detection/Event objects to utilize a single string value for the possibility of adding future tabs.

I went with an outline chevron button down/up because the collapse mdi icon looks way too much like the github clone/download icon to me.

**Comment by subdavis** [comment]
It's Thursday afternoon and I'm feeling persnickety. 

* Make both buttons `depressed` -- the raised shadow looks wrong when everything else in the app is flat.
* I like the icon you chose, but I think the solid variant would fit better.  Aside from the toggles where we switch between solid and outlined (editing mode, star, checkbox), we've used solid icons everywhere else in the app.
* Bring the buttons into alignment with both the leading edge of the scrub bar (in Y) and the center playback controls (in X).   This will also get rid of that thin black line at the bottom of the page when the tabs are collapsed.

Here's what I changed the scss to:

```scss
  position:relative;
  top: -38px;
  left: 2px;
  margin-bottom: -28px;
  width: 250px;
```

**Comment by subdavis** [comment]
That stupid negative margin hack could be solved by adding a slot to `components/Controls.vue`.   That would prevent the entire positioning problem.  But I've since been too lazy to do that myself.

**Comment by BryonLewis** [comment]
I didn't even notice the raised buttons until I looked at it on my laptop screen instead of my monitor for some reason.
Okay implemented the slot mechanism for the tab controls.  Used vuetify's system for centering of the playback controls until the width gets too small then it fits it in the remaining space.

Toggling outline on and off was making the text shift because of the additional border so I added a class to maintain a transparent border when not clicked.  We could also indicate the selected item using something like your accent color although I feel that might be too noticeable for something small.  Open to any opinions on that.

**Pull request merged**